test/test_autograd.py

```diff
@@ -8460,6 +8460,56 @@ def _test_op(fn, inp, args):
         _test_op(torch.view_as_complex, torch.rand(2, 2), ())
         _test_op(torch.view_as_real, torch.rand(2, 2, dtype=torch.cfloat), ())
 
+    def test_setup_context_when_forward_has_default_args(self):
+        class PowFunction(Function):
+            @staticmethod
+            def forward(x, y=3):
+                return torch.pow(x, y)
+
+            @staticmethod
+            def setup_context(ctx, inputs, output):
+                x, y = inputs
+                ctx.save_for_backward(x)
+                ctx.y = y
+
+            @staticmethod
+            def backward(ctx, gO):
+                x, = ctx.saved_tensors
+                y = ctx.y
+                return gO * y * torch.pow(x, y - 1), None
```

Comment:
Can you add a test by adding a new autograd.Function to https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/autograd_function_db.py ?

This will run it through a bunch of tests, including the functorch ones.

Reply:
Of course, have added.